### PR TITLE
fix json.WriteStream.emitJson

### PR DESCRIPTION
**Pull request by xackus**

### CODE REVIEW

File: lib/std/json/write_stream.zig [hunk 1]
```diff
@@ -148,6 +148,7 @@ pub fn WriteStream(comptime OutStream: type, comptime max_depth: usize) type {
             self.popState();
         }
 
+        // TODO better handling of ComptimeInt and ComptimeFloat
```
Comment (LemonBoy) [inline lib/std/json/write_stream.zig:151]: `comptime_int` is quite easy to handle, just cast it to the type returned by `math.IntFittingRange(value, value)` and you can treat it as an int type.

File: lib/std/json/write_stream.zig [hunk 2]
```diff
@@ -191,7 +193,9 @@ pub fn WriteStream(comptime OutStream: type, comptime max_depth: usize) type {
 
         /// Writes the complete json into the output stream
         pub fn emitJson(self: *Self, json: std.json.Value) Stream.Error!void {
+            assert(self.state[self.state_index] == State.Value);
             try self.stringify(json);
+            self.popState();
```
Comment (xackus) [inline lib/std/json/write_stream.zig:198]: The fix is here

Comment (xackus) [reply lib/std/json/write_stream.zig:198]: The other changes are just asserts and improved tests.

File: lib/std/json/write_stream.zig [hunk 3]
```diff
@@ -168,8 +168,11 @@ pub fn WriteStream(comptime OutStream: type, comptime max_depth: usize) type {
                         return;
                     }
                 },
-                .Float => if (@floatCast(f64, value) == value) {
-                    try self.stream.print("{}", .{value});
+                .ComptimeInt => {
+                    return self.emitNumber(@as(std.math.IntFittingRange(value, value), value));
```
Comment (daurnimator) [inline lib/std/json/write_stream.zig:172]: This needlessly limits to what can fit in an `f64`. Better to use: `try self.stringify(value); self.popState()`
You can combine the `.ComptimeFloat` case with this too.

Comment (daurnimator) [reply lib/std/json/write_stream.zig:172]: nevermind, I see that this is the intention for `emitNumber`. Seems a bit awkward, but go ahead and merge I guess

**Pull request merged**